Cython/Compiler/ModuleNode.py

```diff
@@ -282,6 +282,7 @@ def h_entries(entries, api=0, pxd=0):
             h_guard = self.api_name(Naming.h_guard_prefix, env)
             h_code_start.put_h_guard(h_guard)
             h_code_start.putln("")
+            self._put_setup_code(h_code_start, "InitLimitedAPI")
```

Comment:
I think we should split this up and only include the `Py_LIMITED_API` parts in header files, not the automatic `CYTHON_LIMITED_API` definition. If `Py_LIMITED_API` is set, it's a clear sign that the target is the Limited API. `CYTHON_LIMITED_API` should no longer be used for that, and especially not be used (or even needed) by header files.